.ci/metrics/metrics.py

```diff
@@ -178,38 +160,68 @@ def get_per_workflow_metrics(
                 # TODO(boomanaiden154): Remove this once the premerge pipeline is no
                 # longer in a testing state and we can directly assert the workflow
                 # result.
-                for step in workflow_job.steps:
+                for step in job.steps:
                     if step.conclusion != "success" and step.conclusion != "skipped":
                         job_result = 0
                         break
 
+            created_at = job.created_at
+            started_at = job.started_at
+            completed_at = job.completed_at
             queue_time = started_at - created_at
             run_time = completed_at - started_at
-
             if run_time.seconds == 0:
                 continue
 
+            # Grafana will refuse to ingest metrics older than ~2 hours, so we
+            # should avoid sending historical data.
+            metric_age_mn = (
+                datetime.datetime.now(datetime.timezone.utc) - completed_at
+            ).total_seconds() / 60
+            if metric_age_mn > GRAFANA_METRIC_MAX_AGE_MN:
```

Comment:
Can you add some logging here so we can see better when this happens?

Reply:
added logging